tests/lib/rules/no-identical-names.js

```diff
@@ -68,6 +68,18 @@ ruleTester.run("no-identical-title", rule, {
             }
         },
 
+        // Tests with identical names are allowed if they are in different modules.
+        outdent`
+            test('it1', function (assert) {});
+            module('module1', function () {
+                test('it1', function (assert) {});
+                module('module2', function () {
+                    test('it1', function (assert) {});
+                });
+            });
+            test('it1', function (assert) {});
```

Comment:
Should this second test, outside all modules, be flagged? I feel like it should.

Reply:
Yes, it should be flagged, so I marked this PR as a draft since I am going to need to do some refactoring / adding a stack to handle this case.

```js
```js
test("it1", function() {});
module("module1", function() {});
test("it1", function() {}); // This should be a violation.

Reply:
Thanks! Let me know when it's ready for review again or if you need any help.

Reply:
@platinumazure I'm having some second thoughts about this. Technically, the second `test` is part of a separate module, so we should allow it. If we want to disallow top-level tests from having the same title even if they're in different modules, that would require a special case, and I'll likely want to handle that in a separate PR.

Reply:
> Technically, the second `test` is part of a separate module, so we should allow it. If we want to disallow top-level tests from having the same title even if they're in different modules, that would require a special case, and I'll likely want to handle that in a separate PR.

Would you mind elaborating on this? What do you mean the tests are in separate modules?

Reply:
My bad, I was wrong about that, the tests from my last example are in the same module. To clarify, here are two test cases:

Invalid:

```js
test("it1", function() {});
module("module1", function() {});
test("it1", function() {}); // Violation
```

Valid:

```js
test("it1", function() {});
module("module1");
test("it1", function() {}); // This is in module1, so not a violation
```

I have updated the implementation to reflect this.

Reply:
Thanks @bmish for clarifying. I'll give this another review this weekend.